### DISCUSSION TITLE
Covers ASCOM driver

### DISCUSSION THREAD

**Discussion by blumzi** [blumzi/MAST_unit#2] [Ideas]
The **_PlaneWave Shutter Control_** application is configured to use the USB (the `Connection method -> USB` radio button is selected).

The driver seems to be trying to access the device via TCP as well (I can see the failed attempts in the debugger console).

Is there a way to configure it not to try the TCP connections?

It sems to me that the `Connected` propery becomes `False` after being set to `True`, maybe due to it not being able to connect via TCP (not 100% sure :-()

**Comment by kivarsen** [comment]
The PWShutter.exe application can talk to the shutter controller via either USB-Serial or Ethernet (TCP). On the other end. PWShutter.exe listens for commands from other applications, which is documented under the Help menu. The ASCOM driver communicates with PWShutter.exe using this local TCP connection. So you have a few options for controlling the shutters:

- Running PWShutter.exe and talking directly to its TCP server
- Running PWShutter.exe and talking to it via the ASCOM driver
- Talking directly to the mirror cover controller via TCP or USB-Seria

If you want to communicate directly (bypassing PWShutter), I have sample code here:

https://planewave.com/files/tech/pwshutter.py

This depends on the pyserial library (available via pip), and the auxlib library for talking to PlaneWave devices that use our "Aux" protocol, available here:
https://planewave.com/files/tech/auxlib.py
